makefile.shade

```diff
@@ -1348,39 +1328,96 @@ functions @{
         return exitCode;
     }
 
-    HashSet<string> ListDependencyAssemblies(string projectFolder, string framework, string coreclrFolder = null)
+    HashSet<string> ListDependencyAssemblies(string projectFolder, RuntimeTarget target, string ROOT)
     {
-        var tmpFile = Path.GetTempFileName();
-        var argument = string.Format("list {0} --framework {1} --assemblies", projectFolder, framework);
-
-        if (coreclrFolder != null)
+        var runtimeTarget = "";
+        var args = "--runtime {0}";
+        if(target.OS == "win")
         {
-            argument += (" --runtime " + coreclrFolder);
+            runtimeTarget = "win7";
         }
-
-        int result;
-        if (!IsMono)
+        else if(target.OS == "linux")
         {
-            result = ExecuteAndRedirectOutput("cmd", string.Format("/C dnu {0}", argument), tmpFile);
+            runtimeTarget = "ubuntu.14.04";
```

Comment:
Then CoreCLR is technically unsupported on your OS :). As far as I know, they are only targeting the current LTS version: 14.04.

It may, of course, technically work, but it's not a production-supported distro as far as I know.